Add servlets that query and insert data in verifications table

vahinim opened the pull request:
Add SpannerClient.java
Add VerificationDatabaseHelper.java
Add GetStoresVerificationServlet.java, that fetches data from verifications table
Add AddStoreVerificationStatusServlet.java, that inserts verification data into verifications table

vanshikagoogle committed:
Add skeleton (directory structure and model classes) for server

vanshikagoogle committed:
Add javadocs to all classes in models. Renamed fields in Verification.java

vanshikagoogle committed:
Add parameter descriptions for Agent.java, Store.java, Verification.java

vanshikagoogle committed:
Added dependencies for Gson and Spanner to pom.xml

vanshikagoogle committed:
Removed jetty dependency from pom.xml

vanshikagoogle committed:
Fixed ambiguity of variable name verificationCreationDareTime in class Verification.java

vanshikagoogle committed:
Reverted removal of jetty dependency (it is required)

vanshikagoogle committed:
Added getter and setter for store owner name

vahinim committed:
Change Verification.java model

Change AgentID to AgentEmail
Change StoreID to StorePhone

vahinim committed:
Add constant strings representing store verification status in Verification.java model

vahinim committed:
Change NOT_REGISTERED to NOT_VERIFIED

vahinim committed:
Add servlets that insert and query data from verifications table

Add SpannerClient.java
Add VerificationDatabaseHelper.java
Add GetStoresVerificationServlet.java
Add AddStoreVerificationStatusServlet.java

vahinim committed:
Add comments to .java files in VerificationServlet

vahinim committed:
Add another constructor for Agent.java, Store.java, Verification.java

This constructor is used while querying.

vahinim committed:
Merge branch 'backend-skeleton' of https://github.com/googleinterns/smb-fos-tracker into verification_servlets

vahinim committed:
Remove bugs in GetStoreVerificationServlet.java

Remove output argument in query call in GetStoreVerificationServlet.java
Add TimeStamp field using another constructor in Verification.java

vahinim committed:
Add comments and change Mutation object to Mutation list

Add comments to AddStoreVerificationStatusServlet.java
Change Mutation object to Mutation list in writeData() method of VerificationDatabaseHelper.java
Add 403 error messages in AddStoreVerificationStatusServlet.java

vahinim committed:
Remove bug in getStoreVerificationString() method in Verification.java

vahinim committed:
Merge branch 'backend-skeleton' into verification_servlets

vahinim committed:
Add another field ( NEEDS_REVISIT ) in GetStoreVerificationsServlet.java

vahinim committed:
Add another field for verification status in Verification.java

vahinim committed:
Add another field NEEDS_REVISIT in Verification.java

vahinim committed:
Merge branch 'backend-skeleton' into verification_servlets

krati-tiwari commented on FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/AddStoreVerificationStatusServlet.java:
```diff
@@ -0,0 +1,57 @@
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        String json = request.getReader().readLine();
```
rename to verificationRequest?

vahinim replied:
done

krati-tiwari commented on FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/GetStoreVerificationsServlet.java:
```diff
@@ -0,0 +1,91 @@
+            int columnVerificationLatitudeIndex = verificationData.getColumnIndex(VerificationDatabaseHelper.COLUMN_VERIFICATION_LATITUDE);
+            int columnVerificationLongitudeIndex = verificationData.getColumnIndex(VerificationDatabaseHelper.COLUMN_VERIFICATION_LONGITUDE);
+            int columnVerificationStatusIndex = verificationData.getColumnIndex(VerificationDatabaseHelper.COLUMN_VERIFICATION_STATUS);
+            int columnVerificationTimeIndex = verificationData.getColumnIndex(VerificationDatabaseHelper.COLUMN_VERIFICATION_TIME);
```
Make each statement in 2 lines for better readability.

vahinim replied:
done

krati-tiwari commented on FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/VerificationDatabaseHelper.java:
```diff
@@ -0,0 +1,79 @@
+                .set(COLUMN_VERIFICATION_LATITUDE).to(newVerification.getVerificationCoordinates().getLatitude())
+                .set(COLUMN_VERIFICATION_LONGITUDE).to(newVerification.getVerificationCoordinates().getLongitude())
+                .set(COLUMN_VERIFICATION_STATUS).to(newVerification.getStoreVerificationStatus())
+                .set(COLUMN_VERIFICATION_TIME).to(Timestamp.of(java.sql.Timestamp.valueOf(ZonedDateTime.now(ZoneId.of("Asia/Kolkata")).toLocalDateTime())))
```
Split in 2 line

vahinim replied:
done

krati-tiwari commented on FosServer/src/main/java/com/example/fostracker/models/Agent.java:
```diff
@@ -27,6 +29,21 @@ public Agent(Name name, String email, String phone, Coordinates coordinates) {
+     * @param name                  - The agent's name
+     * @param email                 - the gmail address associated with the agent's account
+     * @param phone                 - The phone number associated with the agent's account
+     * @param coordinates           - The live coordinates of the agent at the time of creation. They can be updated later
```
keep same as above params

vahinim replied:
Can you elaborate it a little more ? The second constructor comments are same as first one with last line changed to reflect the agentCreationDateTime argument. And this constructor is used while querying, so we will not set the present time to Agent creation time while querying.

vahinim committed:
Split into two lines and rename to verificationRequest

Split in VerificationDatabaseHelper.java , GetStoreVerificationServlet.java
Rename in AddStoreVerificationStatusServlet.java
Comment in AddStoreVerificationStatusServlet.java

vahinim committed:
Change insert to insertOrUpdate in VerificationDatabaseHelper.java

trblessy commented on FosServer/src/main/java/com/example/fostracker/servlets/HelloAppEngine.java:
```diff
@@ -0,0 +1,34 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "HelloAppEngine", value = "/hello")
```
Do we need this servlet now?

vahinim replied:
removed
